upload/views/bundle_analysis.py

```diff
@@ -116,12 +117,15 @@ def post(self, request: HttpRequest) -> Response:
             },
         )
 
-        upload_external_id = str(uuid.uuid4())
-        storage_path = StoragePaths.upload.path(upload_key=upload_external_id)
-        archive_service = ArchiveService(repo)
-        url = archive_service.storage.create_presigned_put(
-            get_bucket_name(), storage_path, 30
-        )
+        storage_path = data.get("storage_path", None)
+        url = None
+        if storage_path is None or not self.is_shelter_request():
+            upload_external_id = str(uuid.uuid4())
```

Comment:
should pull this var out as it is used in the `reportid` task arguments later on